src/uct/ib/base/ib_iface.h

```diff
@@ -357,4 +359,53 @@ void uct_ib_iface_set_max_iov(uct_ib_iface_t *iface, size_t max_iov)
 }
 
 
+static UCS_F_ALWAYS_INLINE
+void uct_ib_iface_fill_ah_attr_from_gid_lid(uct_ib_iface_t *iface, uint16_t lid,
+                                            const union ibv_gid *gid,
+                                            uint8_t path_bits,
+                                            struct ibv_ah_attr *ah_attr)
+{
+    memset(ah_attr, 0, sizeof(*ah_attr));
+
+    ah_attr->sl                = iface->config.sl;
+    ah_attr->src_path_bits     = path_bits;
+    ah_attr->dlid              = lid | path_bits;
+    ah_attr->port_num          = iface->config.port_num;
+    ah_attr->grh.traffic_class = iface->config.traffic_class;
+
+    if ((gid != NULL) &&
+        ((iface->addr_type == UCT_IB_ADDRESS_TYPE_ETH)    ||
+         (iface->addr_type == UCT_IB_ADDRESS_TYPE_GLOBAL) ||
+         (iface->gid.global.subnet_prefix != gid->global.subnet_prefix))) {
+        ah_attr->is_global = 1;
+        ah_attr->grh.sgid_index = iface->config.gid_index;
+        ah_attr->grh.dgid = *gid;
+    } else {
+        ah_attr->is_global = 0;
+    }
+}
+
+static UCS_F_ALWAYS_INLINE
+void uct_ib_iface_fill_ah_attr_from_addr(uct_ib_iface_t *iface,
+                                         const uct_ib_address_t *ib_addr,
+                                         uint8_t path_bits,
+                                         struct ibv_ah_attr *ah_attr)
+{
+    union ibv_gid *gid_p = NULL;
+    union ibv_gid  gid;
+    uint8_t        is_global;
+    uint16_t       lid;
+
+    uct_ib_address_unpack(ib_addr, &lid, &is_global, &gid);
+
+    /* Create a global address only if we cannot reach the destination using local
```

Comment:
this comment is not correct anymore, and needs to be removed. we use global address for every destination which has either ETH or ADDR_TYPE=ib_global, whether or not on same subnet.

Reply:
removed